optuna/samplers/_grid.py

```diff
@@ -211,7 +211,11 @@ def _get_unvisited_grid_ids(self, study: Study) -> List[int]:
 
         # List up unvisited grids based on already finished ones.
         visited_grids = []
-        for t in study.trials:
+
+        # When `HyperbandPruner` is used, `study.get_trials` returns filtered trials.
```

Comment:
I think this can be more descriptive. How about the following?
```suggestion
        # We directly query the storage to get trials here instead of `study.get_trials`,
        # since some pruners such as `HyperbandPruner` use the study transformed
        # to filter trials. See https://github.com/optuna/optuna/issues/2327 for details.
```